src/utils/scalar_functions.jl

```diff
@@ -28,7 +28,6 @@ struct QuadraticControlFunction{T, MT <: AbstractMatrix{T}} <: ScalarFunction
     Q::MT
 end
 function function_value(f::QuadraticControlFunction, x)
```

Comment:
You can probably still put <: ScalarFunction on line 30 to specify output type right?

Reply:
but the output is float vector